topdown/walk.go

```diff
@@ -5,9 +5,32 @@
 package topdown
 
 import (
+	"sync"
+
 	"github.com/open-policy-agent/opa/ast"
 )
 
+var (
+	emptyArr = ast.ArrayTerm()
+
+	arrayPool = sync.Pool{
+		New: func() any {
+			return ast.NewArray(
+				// These values don't matter as they'll be immediately replaced.
+				// Used only to initialize the size of the array.
+				ast.InternedBooleanTerm(false),
+				ast.InternedBooleanTerm(false),
+			)
+		},
+	}
+
+	termPool = sync.Pool{
+		New: func() any {
+			return ast.NewTerm(ast.Boolean(false))
```

Comment:
I think I'd be less surprised about something like `return &ast.Term{}` -- why bother with a Value if it's not supposed to have one in the first place?

Reply:
`ast.NewTerm(nil)` would also be slightly less confusing, imho.

Reply:
Sure, I can go with that